envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/EnvoyListenersFactory.kt

```diff
@@ -49,6 +55,41 @@ class EnvoyListenersFactory(
     private val egressRdsInitialFetchTimeout: Duration = durationInSeconds(20)
     private val ingressRdsInitialFetchTimeout: Duration = durationInSeconds(30)
 
+    private val tlsAuthenticationProperties = snapshotProperties.incomingPermissions.tlsAuthentication
+    private val requireClientCertificate = BoolValue.of(tlsAuthenticationProperties.requireClientCertificate)
+    private val downstreamTlsContext = DownstreamTlsContext.newBuilder()
+            .setRequireClientCertificate(requireClientCertificate)
+            .setCommonTlsContext(CommonTlsContext.newBuilder()
+                    .setTlsParams(TlsParameters.newBuilder()
+                            .setTlsMinimumProtocolVersion(TlsParameters.TlsProtocol.TLSv1_2)
```

Comment:
IMO Tls version protocol and Cipher should be configurable.

Reply:
That might be a good idea, but I would point to our performance tests. Has @MarcinFalkowski published it somewhere?

Reply:
No, they are not published. We should run them again, maybe TLS 1.3 performance is fixed now.
Anyway, I think it is a good idea to extract tls parameters to the properties.